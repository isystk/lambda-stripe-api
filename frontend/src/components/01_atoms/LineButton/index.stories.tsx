import { Meta, Story } from '@storybook/react'
import React from 'react'
import LineButton from './index'

export default {
  title: '01_atoms/LineButton',
  component: LineButton,
} as Meta

const Template: Story = (props) => {
  return <LineButton link="#" label="友達に追加して質問してみる" />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
