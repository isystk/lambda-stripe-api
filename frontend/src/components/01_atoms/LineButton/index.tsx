import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** LineButtonProps Props */
export type LineButtonProps = WithChildren & { link; label }
/** Presenter Props */
export type PresenterProps = LineButtonProps

/** Presenter Component */
const LineButtonPresenter: FC<PresenterProps> = ({
  link = '#',
  label = '',
}) => (
  <>
    <a
      href={link}
      className={`${styles.lineButton} p-2 pl-4 pr-4 flex items-center justify-center bg-green-500 rounded-full`}
    >
      <img src="/images/icon-line.png" width="50px" height="50px"></img>
      <span className="text-white font-bold">{label}</span>
    </a>
  </>
)

/** Container Component */
const LineButtonContainer: React.FC<
  ContainerProps<LineButtonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  return presenter({
    children,
    ...props,
  })
}

export default connect<LineButtonProps, PresenterProps>(
  'LineButton',
  LineButtonPresenter,
  LineButtonContainer
)
