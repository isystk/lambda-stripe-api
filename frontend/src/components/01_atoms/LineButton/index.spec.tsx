import React from 'react'
import renderer from 'react-test-renderer'
import LineButton from './index'
import '@testing-library/jest-dom/extend-expect'

describe('LineButton', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<LineButton name="sample" />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
