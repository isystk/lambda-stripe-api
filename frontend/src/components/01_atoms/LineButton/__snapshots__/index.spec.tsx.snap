// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineButton Match Snapshot 1`] = `
<a
  className="css-0 p-2 pl-4 pr-4 flex items-center justify-center bg-accent rounded-full"
  href="#"
>
  <img
    alt=""
    height="50px"
    src="/images/icon-line.png"
    width="50px"
  />
  <span
    className="text-white font-bold"
  />
</a>
`;
